### PR TITLE
Decrease package sizes

### PR DESCRIPTION
**Pull request by bennodev19**
## Description
Decreased pulse package size.. by only building the dist folder and not the whole folder with tests, ..
See in agileTs which is basically the same as pulseJs but has only uploaded the dist folder to npm
-> pulse would have a package size of about 150kb instead of 800kb
https://www.npmjs.com/package/@agile-ts/core
https://www.npmjs.com/package/@pulsejs/core

## How Has This Been Tested?
Couldn't be tested since its npm related

### CODE REVIEW

File: docs/v3/getting-started/setup-with-react.md [hunk 1]
```diff
@@ -49,15 +53,16 @@ export default function Component(): React.FC {
...
 
 usePulse also supports **arrays** of State instances, returning values as an array that can be destructured.
 
 ```ts
 const [myState, anotherState] = usePulse([core.MY_STATE, core.ANOTHER_STATE]);
 ```
- The names of the values can be anything, though we recommend they be the camel case counter-part to the State instance name. This is completely typesafe as of version `3.1`
-  
+
+The names of the values can be anything, though we recommend they be the camel case counter-part to the State instance name. This is completely typesafe as of version `3.1`
```
Comment (hadenpf) [inline docs/v3/getting-started/setup-with-react.md:65]: Would it be possible to put docs modifications in a separate PR, or otherwise please make a note of them in the primary PR comment?

File: packages/pulse-core/package.json [hunk 2]
```diff
@@ -17,5 +17,8 @@
     "type": "git",
     "url": "https://github.com/pulse-framework/pulse"
   },
-  "gitHead": "df2d02d8e5a07ded7bb792c0d8e1473c5b6159ba"
+  "gitHead": "df2d02d8e5a07ded7bb792c0d8e1473c5b6159ba",
```
Comment (hadenpf) [inline packages/pulse-core/package.json:20]: This isn't your fault, but a note for us — gitHead [is not supposed to be versioned](https://github.com/lerna/lerna/issues/1880#issuecomment-456160357) into the repo. I'll remove all instances of it before merging and find out why it got committed.

File: packages/pulse-next/package.json [hunk 3]
```diff
@@ -25,5 +25,8 @@
     "type": "git",
     "url": "https://github.com/pulse-framework/pulse"
   },
-  "gitHead": "df2d02d8e5a07ded7bb792c0d8e1473c5b6159ba"
+  "gitHead": "df2d02d8e5a07ded7bb792c0d8e1473c5b6159ba",
+  "files": [
+    "dist/**/*"
```
Comment (hadenpf) [inline packages/pulse-next/package.json:30]: As far as I'm aware, `files` is supposed to be an exhaustive list of input files — `include` should be used for any nonspecific paths (such as the recursive glob pattern here). I'll have to run this by when I test.

**Review by hadenpf — CHANGES_REQUESTED** [review]
Overall, good PR! I'll run some tests on this as soon as I have a chance, but it's a great idea & if it all works as intended then it's great!

File: docs/v3/getting-started/setup-with-react.md (hunk 1 above)
Comment (bennodev19) [reply docs/v3/getting-started/setup-with-react.md:65]: I thinks this changes are from jamie.. I only changed the tsconfig.json files and package.json files.. 
maybe I did something wrong by rebasing the up to date master to my master

File: packages/pulse-next/package.json (hunk 3 above)
Comment (bennodev19) [reply packages/pulse-next/package.json:30]: I don't know what the advantage of 'include' is I only know that it worked with 'files' for agileTs quite fine..

Comment (bennodev19) [reply packages/pulse-next/package.json:30]: * Edit.. Idk if include workes at all.. 
I tested it with ` "include ": "dist/**/*"` and it didn't exclude the tests and lib

This is the link how I got to 'files':
https://remarkablemark.org/blog/2017/09/23/npm-publish-include-files/

Comment (hadenpf) [reply packages/pulse-next/package.json:30]: Ahh, I confused package.json for tsconfig.json! I'm sorry!

File: docs/v3/getting-started/setup-with-react.md (hunk 1 above)
Comment (hadenpf) [reply docs/v3/getting-started/setup-with-react.md:65]: It could be that — I did change the base of the PR to develop rather than master, since that's where unreleased code is committed before a release is made. Could be entirely my fault! I'm sorry if so!

Comment (bennodev19) [reply docs/v3/getting-started/setup-with-react.md:65]: sure.. next time I will pull request into develop ;D
